---
 configure.ac |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- libunwind-1.1.orig/configure.ac
+++ libunwind-1.1/configure.ac
@@ -135,11 +135,11 @@ AM_CONDITIONAL(OS_HPUX, expr x$target_os
 AM_CONDITIONAL(OS_FREEBSD, expr x$target_os : xfreebsd >/dev/null)
 
 AC_MSG_CHECKING([for ELF helper width])
 case "${target_arch}" in
 (arm|hppa|ppc32|x86|sh) use_elf32=yes; AC_MSG_RESULT([32]);;
-(ia64|ppc64|x86_64)    use_elf64=yes; AC_MSG_RESULT([64]);;
+(ia64|ppc64|x86_64|aarch64)    use_elf64=yes; AC_MSG_RESULT([64]);;
 (mips)                 use_elfxx=yes; AC_MSG_RESULT([xx]);;
 *)                     AC_MSG_ERROR([Unknown ELF target: ${target_arch}])
 esac
 AM_CONDITIONAL(USE_ELF32, [test x$use_elf32 = xyes])
 AM_CONDITIONAL(USE_ELF64, [test x$use_elf64 = xyes])
