---
 CMake/HPHPFindLibs.cmake |   22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

--- git.orig/CMake/HPHPFindLibs.cmake
+++ git/CMake/HPHPFindLibs.cmake
@@ -43,12 +43,12 @@ find_package(Libinotify)
 if (LIBINOTIFY_INCLUDE_DIR)
 	include_directories(${LIBINOTIFY_INCLUDE_DIR})
 endif()
 
 # unwind checks
-find_package(Libunwind REQUIRED)
-include_directories(${LIBUNWIND_INCLUDE_DIR})
+# find_package(Libunwind REQUIRED)
+# include_directories(${LIBUNWIND_INCLUDE_DIR})
 
 # iconv checks
 find_package(Libiconv REQUIRED)
 include_directories(${LIBICONV_INCLUDE_DIR})
 if (LIBICONV_CONST)
