diff -urN acpica-unix-20140114/tests/aapits/Makefile acpica-unix-20140114.patch/tests/aapits/Makefile
--- acpica-unix-20140114/tests/aapits/Makefile	2014-01-14 15:02:45.000000000 -0700
+++ acpica-unix-20140114.patch/tests/aapits/Makefile	2014-01-23 16:07:50.124972105 -0700
@@ -29,6 +29,7 @@
 	../../source/components/debugger/dbmethod.c \
 	../../source/components/debugger/dbnames.c \
 	../../source/components/debugger/dbstats.c \
+	../../source/components/debugger/dbtest.c \
 	../../source/components/debugger/dbutils.c \
 	../../source/components/debugger/dbxface.c \
 	../../source/components/disassembler/dmbuffer.c \

