Index: acpica-unix/Makefile
===================================================================
--- acpica-unix.orig/Makefile	2013-01-31 15:25:49.000000000 -0700
+++ acpica-unix/Makefile	2013-01-31 15:25:49.000000000 -0700
@@ -14,3 +14,25 @@
 
 include generate/unix/Makefile.config
 include generate/unix/Makefile.common
+
+check:
+	# ASL tests
+	$(CURDIR)/tests/aslts.sh $(CURDIR)/tests/aslts $(CURDIR)/generate/unix
+
+	# API Tests
+	$(CURDIR)/debian/run-aapits.sh $(CURDIR)/tests/aapits $(CURDIR)/generate/unix/bin
+
+	# misc tests
+	$(CURDIR)/debian/run-misc-tests.sh $(CURDIR) 20131115
+
+	# Template tests
+	cd $(CURDIR)/tests/templates
+	make
+	if [ -f diff.log ] ; \
+	then \
+		if [ -s diff.log ] ; \
+		then \
+			exit 1		# implies errors occurred ; \
+		fi ; \
+	fi
+
