Index: gdb-linaro-7.6.1-2013.10/bfd/elf64-aarch64.c
===================================================================
--- gdb-linaro-7.6.1-2013.10.orig/bfd/elf64-aarch64.c
+++ gdb-linaro-7.6.1-2013.10/bfd/elf64-aarch64.c
@@ -1529,7 +1529,7 @@ elf64_aarch64_grok_prstatus (bfd *abfd,
       default:
 	return FALSE;
 
-      case 408:		/* sizeof(struct elf_prstatus) on Linux/arm64.  */
+      case 392:		/* sizeof(struct elf_prstatus) on Linux/arm64.  */
 	/* pr_cursig */
 	elf_tdata (abfd)->core->signal
 	  = bfd_get_16 (abfd, note->descdata + 12);
