--- a/harness/cases/16.t
+++ b/harness/cases/16.t
@@ -3,25 +3,7 @@
 */
 #include <stdint.h>
 #include <err.h>
-#include <sys/syscall.h>   /* For SYS_xxx definitions */
-
-#ifndef SYS_eventfd
-#if defined(__i386__)
-#define SYS_eventfd 323
-#elif defined(__x86_64__)
-#define SYS_eventfd 284
-#elif defined(__ia64__)
-#define SYS_eventfd 1309
-#elif defined(__PPC__)
-#define SYS_eventfd 307
-#elif defined(__s390__)
-#define SYS_eventfd 318
-#elif defined(__alpha__)
-#define SYS_eventfd 478
-#else
-#error define SYS_eventfd for your arch!
-#endif
-#endif
+#include <sys/eventfd.h>
 
 int test_main(void)
 {
@@ -39,7 +21,7 @@
 	struct timespec	notime = { .tv_sec = 0, .tv_nsec = 0 };
 
 	buf = malloc(SIZE);				assert(buf);
-	efd = syscall(SYS_eventfd, 0);
+	efd = eventfd(0, 0);
 	if (efd < 0) {
 		if (errno == ENOSYS) {
 			printf("No eventfd support.  [SKIPPING]\n");
