Index: ltp-full-20130904/testcases/kernel/include/aarch64.in
===================================================================
--- ltp-full-20130904.orig/testcases/kernel/include/aarch64.in	2013-09-04 10:18:22.000000000 +0300
+++ ltp-full-20130904/testcases/kernel/include/aarch64.in	2013-10-11 11:20:47.565921406 +0300
@@ -253,5 +253,3 @@
 syncfs 267
 setns 268
 sendmmsg 269
-cacheflush 0x1002
-set_tls 0x1005
Index: ltp-full-20130904/testcases/kernel/include/linux_syscall_numbers.h
===================================================================
--- ltp-full-20130904.orig/testcases/kernel/include/linux_syscall_numbers.h	2013-09-04 10:18:22.000000000 +0300
+++ ltp-full-20130904/testcases/kernel/include/linux_syscall_numbers.h	2013-10-11 11:22:20.013925880 +0300
@@ -801,12 +801,6 @@
 # ifndef __NR_sendmmsg
 #  define __NR_sendmmsg 269
 # endif
-# ifndef __NR_cacheflush
-#  define __NR_cacheflush 0x1002
-# endif
-# ifndef __NR_set_tls
-#  define __NR_set_tls 0x1005
-# endif
 #endif
 
 
@@ -12754,9 +12748,6 @@
 # ifndef __NR_settimeofday
 #  define __NR_settimeofday __LTP__NR_INVALID_SYSCALL
 # endif
-# ifndef __NR_set_tls
-#  define __NR_set_tls __LTP__NR_INVALID_SYSCALL
-# endif
 # ifndef __NR_setuid
 #  define __NR_setuid __LTP__NR_INVALID_SYSCALL
 # endif
