Index: ltp-full-20130904/testcases/kernel/power_management/get_sched_values.c
===================================================================
--- ltp-full-20130904.orig/testcases/kernel/power_management/get_sched_values.c	2013-09-04 10:18:22.000000000 +0300
+++ ltp-full-20130904/testcases/kernel/power_management/get_sched_values.c	2013-10-11 14:57:40.890551214 +0300
@@ -17,6 +17,8 @@
 #include <stdio.h>
 #include "test.h"
 
+const char *TCID = "Power_Management";
+
 int get_supp_sched_mc(void)
 {
 	if (tst_kvercmp(2, 6, 29) < 0)
